frontend/src/components/Panel.svelte

```diff
@@ -0,0 +1,94 @@
+<script lang="ts">
+
+    import GraphOptions from "./GraphOptions.svelte";
+    import ColorLegend from "./ColorLegend.svelte";
+    import EventDetail from "./EventDetail.svelte";
+    import QueryForm from "./QueryForm.svelte";
+    import type { GraphSettings } from "../layout";
+    import type { FixedQuery } from "../uitypes";
+    import type { FullEvent } from "../apidefinition";
+    
+    //Boolean variables
+    export let show_legend_placeholder: boolean; 
+    export let show_menu_placeholder: boolean; 
+    export let awaiting_query_request: boolean; 
+    export let current_query_changed: boolean; 
+
+    //Object variables; used for onClick actions.
+    export let reset_graph_options_placeholder: any; 
+    export let consume_query: any; 
+    export let use_selected_as_root: any; 
+    export let add_filter: any; 
+    export let submit_state_query: any; 
```

Comment:
Type any needs to be avoided if not explicitly explained why it is being used.

Reply:
So some of these methods return Objects, but for reset_graph_options_placeholder for instance, even Object does not work. So I will add a comment about it.